--- origsrc/llvm-10.0.1.src/tools/polly/lib/CMakeLists.txt	2023-07-23 13:48:49.051377600 +0800
+++ src/llvm-10.0.1.src/tools/polly/lib/CMakeLists.txt	2023-07-23 13:43:52.721688100 +0800
@@ -21,7 +21,7 @@
   file(GLOB_RECURSE POLLY_HEADER_FILES "${POLLY_SOURCE_DIR}/include/polly/*.h")
 endif ()
 
-set(POLLY_COMPONENTS
+set(LLVM_LINK_COMPONENTS
     Support
     Core
     ScalarOpts
@@ -46,7 +46,7 @@
 # Polly-ACC requires the NVPTX backend to work. Ask LLVM about its libraries.
 if (GPU_CODEGEN)
   # This call emits an error if they NVPTX backend is not enable.
-  list(APPEND POLLY_COMPONENTS NVPTX)
+  list(APPEND LLVM_LINK_COMPONENTS NVPTX)
 endif ()
 
 # Use an object-library to add the same files to multiple libs without requiring
@@ -99,9 +99,6 @@
   Transform/RewriteByReferenceParameters.cpp
   Transform/ScopInliner.cpp
   ${POLLY_HEADER_FILES}
-
-  LINK_COMPONENTS
-  ${POLLY_COMPONENTS}
   )
 set_target_properties(obj.Polly PROPERTIES FOLDER "Polly")
 set_target_properties(Polly PROPERTIES FOLDER "Polly")
