--- origsrc/llvm-10.0.1.src/CMakeLists.txt	2020-07-08 00:21:37.000000000 +0800
+++ src/llvm-10.0.1.src/CMakeLists.txt	2023-07-23 00:48:52.663108000 +0800
@@ -52,7 +52,11 @@
 
 set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++ standard to conform to")
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
-set(CMAKE_CXX_EXTENSIONS NO)
+if (CYGWIN)
+  set(CMAKE_CXX_EXTENSIONS ON)
+else()
+  set(CMAKE_CXX_EXTENSIONS NO)
+endif()
 
 if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   message(STATUS "No build type selected, default to Debug")
