--- origsrc/llvm-3.0.src/bindings/ocaml/llvm/Makefile	2011-10-14 15:38:02.000000000 -0500
+++ src/llvm-3.0.src/bindings/ocaml/llvm/Makefile	2012-01-21 21:54:31.379593600 -0600
@@ -30,7 +30,7 @@ copy-meta: $(OcamlDir)/META.llvm
 $(OcamlDir)/META.llvm: META.llvm
 	$(Verb) $(CP) -f $< $@
 
-install-meta:: $(ObjDir)/META.llvm
+install-meta:: $(OcamlDir)/META.llvm
 	$(Echo) "Install $(BuildMode) $(DestMETA)"
 	$(Verb) $(MKDIR) $(PROJ_libocamldir)
 	$(Verb) $(DataInstall) META.llvm "$(DestMETA)"
