--- origsrc/llvm-3.0.src/tools/llvm-config/llvm-config.in.in	2010-11-14 13:10:47.000000000 -0600
+++ src/llvm-3.0.src/tools/llvm-config/llvm-config.in.in	2012-01-24 04:16:10.042800200 -0600
@@ -115,7 +115,7 @@ foreach my $arg (@ARGV) {
         } elsif ($arg eq "--ldflags") {
             $has_opt = 1; print "-L$LIBDIR $LDFLAGS $SYSTEM_LIBS\n";
         } elsif ($arg eq "--libs") {
-            $has_opt = 1; $want_libs = 1;
+            $has_opt = 1; print "-lLLVM-$VERSION\n";
         } elsif ($arg eq "--libnames") {
             $has_opt = 1; $want_libnames = 1;
         } elsif ($arg eq "--libfiles") {
