--- origsrc/llvm-3.4.2.src/Makefile.rules	2014-05-15 15:56:00.000000000 -0500
+++ src/llvm-3.4.2.src/Makefile.rules	2014-07-18 00:00:53.210529100 -0500
@@ -1109,6 +1109,9 @@ endif
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
 # LLVMLibsOptions is invalidated at processing tools/llvm-shlib.
 SharedLinkOptions += $(NativeExportsFile)
+ifndef LOADABLE_MODULE
+SharedLinkOptions += -Wl,--out-implib,$(SharedLibDir)/lib$(LIBRARYNAME).dll.a
+endif
 endif
 
 endif
@@ -1204,6 +1207,11 @@ ifdef SHARED_ALIAS
 	$(Echo) Creating alias from $(DestSharedLib) to $(DestSharedAlias)
 	$(Verb) $(AliasTool) $(BaseLibName.SO) $(DestSharedAlias)
 endif
+ifndef LOADABLE_MODULE
+ifeq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
+	$(Verb) $(INSTALL) $(SharedLibDir)/lib$(LIBRARYNAME).dll.a $(DESTDIR)$(PROJ_libdir)
+endif
+endif
 
 uninstall-local::
 	$(Echo) Uninstalling $(BuildMode) Shared Library $(DestSharedLib)
