--- origsrc/llvm-3.4.2.src/tools/llvm-config/llvm-config.cpp	2013-06-26 08:54:34.000000000 -0500
+++ src/llvm-3.4.2.src/tools/llvm-config/llvm-config.cpp	2014-07-18 00:15:07.389996100 -0500
@@ -280,7 +280,7 @@ int main(int argc, char **argv) {
         OS << "-L" << ActiveLibDir << ' ' << LLVM_LDFLAGS
            << ' ' << LLVM_SYSTEM_LIBS << '\n';
       } else if (Arg == "--libs") {
-        PrintLibs = true;
+        OS << "-lLLVM-" << PACKAGE_VERSION << '\n';
       } else if (Arg == "--libnames") {
         PrintLibNames = true;
       } else if (Arg == "--libfiles") {
