--- origsrc/llvm-3.4.2.src/tools/clang/bindings/python/clang/cindex.py	2013-10-30 19:03:33.000000000 -0500
+++ src/llvm-3.4.2.src/tools/clang/bindings/python/clang/cindex.py	2014-07-18 01:42:09.886168900 -0500
@@ -3350,12 +3350,14 @@ class Config:
         if Config.library_file:
             return Config.library_file
 
-        import platform
-        name = platform.system()
+        import sys
+        name = sys.platform
 
-        if name == 'Darwin':
+        if name == 'darwin':
             file = 'libclang.dylib'
-        elif name == 'Windows':
+        elif name == 'cygwin':
+            file = 'cygclang.dll'
+        elif name == 'win32':
             file = 'libclang.dll'
         else:
             file = 'libclang.so'
