--- origsrc/llvm-3.8.0.src/tools/llvm-shlib/CMakeLists.txt	2015-11-04 17:11:12.000000000 -0600
+++ src/llvm-3.8.0.src/tools/llvm-shlib/CMakeLists.txt	2016-06-03 18:13:47.164854400 -0500
@@ -41,7 +41,7 @@ endif()
 add_llvm_library(LLVM SHARED DISABLE_LLVM_LINK_LLVM_DYLIB SONAME ${SOURCES})
 
 list(REMOVE_DUPLICATES LIB_NAMES)
-if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux") # FIXME: It should be "GNU ld for elf"
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR CYGWIN) # FIXME: It should be "GNU ld"
   # GNU ld doesn't resolve symbols in the version script.
   set(LIB_NAMES -Wl,--whole-archive ${LIB_NAMES} -Wl,--no-whole-archive)
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
