--- origsrc/llvm-4.0.1.src/tools/llvm-shlib/CMakeLists.txt	2016-11-01 15:19:33.000000000 -0500
+++ src/llvm-4.0.1.src/tools/llvm-shlib/CMakeLists.txt	2017-06-23 18:07:52.942624000 -0500
@@ -37,7 +37,7 @@ endif()
 add_llvm_library(LLVM SHARED DISABLE_LLVM_LINK_LLVM_DYLIB SONAME ${SOURCES})
 
 list(REMOVE_DUPLICATES LIB_NAMES)
-if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR MINGW) # FIXME: It should be "GNU ld for elf"
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR CYGWIN OR MINGW) # FIXME: It should be "GNU ld"
   # GNU ld doesn't resolve symbols in the version script.
   set(LIB_NAMES -Wl,--whole-archive ${LIB_NAMES} -Wl,--no-whole-archive)
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
