--- origsrc/llvm-4.0.1.src/tools/llvm-config/llvm-config.cpp	2017-01-12 13:47:38.000000000 -0600
+++ src/llvm-4.0.1.src/tools/llvm-config/llvm-config.cpp	2017-06-25 20:52:10.240141500 -0500
@@ -355,6 +355,7 @@ int main(int argc, char **argv) {
     if (HostTriple.isOSCygMing()) {
       StaticExt = "a";
       StaticPrefix = "lib";
+      SharedPrefix = "cyg";
     } else {
       StaticExt = "lib";
       DirSep = "\\";
@@ -416,7 +417,7 @@ int main(int argc, char **argv) {
   /// extension. Returns true if Lib is in a recognized format.
   auto GetComponentLibraryNameSlice = [&](const StringRef &Lib,
                                           StringRef &Out) {
-    if (Lib.startswith("lib")) {
+    if (Lib.startswith("lib") || Lib.startswith("cyg")) {
       unsigned FromEnd;
       if (Lib.endswith(StaticExt)) {
         FromEnd = StaticExt.size() + 1;
