--- origsrc/llvm-6.0.0.src/lib/Testing/Support/CMakeLists.txt	2017-10-27 16:12:28.000000000 -0500
+++ src/llvm-6.0.0.src/lib/Testing/Support/CMakeLists.txt	2018-04-29 22:23:54.201080400 -0500
@@ -4,6 +4,9 @@ add_definitions(-DGTEST_HAS_TR1_TUPLE=0)
 add_llvm_library(LLVMTestingSupport
   Error.cpp
 
+  LINK_COMPONENTS
+  Support
+
   BUILDTREE_ONLY
   
   ADDITIONAL_HEADER_DIRS
--- origsrc/llvm-6.0.0.src/tools/llvm-shlib/CMakeLists.txt	2017-07-19 11:07:51.000000000 -0500
+++ src/llvm-6.0.0.src/tools/llvm-shlib/CMakeLists.txt	2018-04-29 22:23:30.176762600 -0500
@@ -37,7 +37,7 @@ endif()
 add_llvm_library(LLVM SHARED DISABLE_LLVM_LINK_LLVM_DYLIB SONAME ${SOURCES})
 
 list(REMOVE_DUPLICATES LIB_NAMES)
-if(("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux") OR (MINGW) OR (HAIKU)
+if(("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux") OR (CYGWIN) OR (MINGW) OR (HAIKU)
    OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
    OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "DragonFly")
    OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "SunOS")) # FIXME: It should be "GNU ld for elf"
