--- origsrc/llvm-8.0.1.src/tools/polly/CMakeLists.txt	2020-04-28 15:46:10.539586500 -0400
+++ src/llvm-8.0.1.src/tools/polly/CMakeLists.txt	2020-04-28 17:21:35.200000400 -0400
@@ -126,6 +126,9 @@ if (NOT POLLY_BUNDLED_ISL)
   find_package(ISL MODULE REQUIRED)
   message(STATUS "Using external libisl ${ISL_VERSION} in: ${ISL_PREFIX}")
   set(ISL_TARGET ISL)
+  set(ISL_INCLUDE_DIRS
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/polly
+  )
 else()
   set(ISL_INCLUDE_DIRS
     ${CMAKE_CURRENT_BINARY_DIR}/lib/External/isl/include
--- origsrc/llvm-8.0.1.src/tools/polly/cmake/FindISL.cmake	2017-02-27 12:54:25.000000000 -0500
+++ src/llvm-8.0.1.src/tools/polly/cmake/FindISL.cmake	2020-04-28 17:29:49.449896200 -0400
@@ -15,7 +15,7 @@ foreach (libname IN LISTS ISL_LIBRARIES)
       list(APPEND hints  ${dir})
     endforeach()
   endif()
-  find_library(lib NAMES ${libname} HINTS ${hints} NO_DEFAULT_PATH)
+  find_library(lib NAMES ${libname} HINTS ${hints})
   set_property(TARGET ISL APPEND PROPERTY INTERFACE_LINK_LIBRARIES ${lib})
 endforeach()
 
